Sources/FoundationEssentials/URL/URL.swift

```diff
@@ -1146,9 +1146,32 @@ public struct URL: Equatable, Sendable, Hashable {
         }
     }
 
+    private static func windowsPath(for posixPath: String) -> String {
+        let utf8 = posixPath.utf8
+        guard utf8.count >= 4 else {
+            return posixPath
+        }
+        // "C:\" is standardized to "/C:/" on initialization
+        let array = Array(utf8)
+        if array[0] == ._slash,
+           array[1].isAlpha,
+           array[2] == ._colon,
+           array[3] == ._slash {
+            return String(Substring(utf8.dropFirst()))
+        }
```

Comment:
Sorry I don't think I understand. Do we need to strip leading slashes from `\\server\share`? A URL like `URL(filePath: "\\\\server\\share")` is stored internally as `file:////server/share`, so `URL.fileSystemPath` will return `//server/share` in that case.

Reply:
Okay, I think that we can get away with this then. The root classification is a bit tricky as `[A-Z]:` is _not_ a root, `[A-Z]:\` _is_ a root, `\\server` is not a root, but `\\server\share` is a root.

Reply:
Oh that's interesting, I wasn't aware of the `\\server` case. There's likely room for improvement on how we initialize a URL like `URL(filePath: "\\server")` on Windows. Currently, this is treated as absolute (`.absoluteString == "file:////server", .path == "//server"`), but it sounds like this might be relative instead?

I don't think it blocks this PR to strip the leading slash from Windows drive letter RFC paths, but it's something we should consider after.